databroker/core.py

```diff
@@ -647,7 +650,28 @@ def stream_gen(entry):
         streams = [stream_gen(entry) for entry in self._entries.values()]
 
         yield ('start', self.metadata['start'])
-        yield from interlace(*streams, strict_order=strict_order)
+
+        # This following code filters out duplicate documents.
+        # This is needed because we dont know which EventStream, that resource
+        # or datum documents belong to, so each stream has these documents.
+        # Without this filter we would get multiple of the same resource and
+        # datum documents.
+        for name, doc in interlace(*streams, strict_order=strict_order):
+
+            if name == 'datum':
```

Comment:
Could this be moved into `interlace` itself to avoid code duplication?

Reply:
I think its better if interlace is just for interlacing, and not also filtering. I kind of think it belongs in canonical. All of canonical is duplicated. What if I fix the duplication by getting rid of the duplication of canonical?

Reply:
That's a great suggestion. (Thanks for your patience with my slow reply.)